### PR TITLE
ref(spans): Add support for scoped metadata to otel conversion

### PR DESCRIPTION
**Pull request by Dav1dde**
Add support for otel scoped metadata to span conversion (like we already have for logs).

This already is hacked in when converting otel traces to 'OtelSpan', will clean this up once we remove the `OtelSpan` type.

### CODE REVIEW

File: relay-spans/src/otel_to_sentry.rs [hunk 1]
```diff
@@ -107,7 +113,7 @@ mod tests {
             "droppedLinksCount": 0
         }"#;
         let otel_span: OtelSpan = serde_json::from_str(json).unwrap();
-        let event_span: EventSpan = otel_to_sentry_span(otel_span);
+        let event_span: EventSpan = otel_to_sentry_span(otel_span, None, None);
```
Comment (jjbayer) [inline relay-spans/src/otel_to_sentry.rs]: Should we add a test where these are not `None`?

**Review by jjbayer — APPROVED** [review]
